### PR TITLE
Add Google Programmable Search Engine to 1-Sidebar and 2-404 page

### PR DESCRIPTION
**Pull request by loricooperhdf**
I'm not sure how this will render. (Internal JIRA ticket: https://hdfgroup.atlassian.net/browse/IWIP-544) 

We haven't run this through committee. However, I think it's valid and important. (#YOLO.)

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

File: 404.md [hunk 1]
```diff
@@ -5,4 +5,10 @@ title: 404 - Content Not Found
 
 # This page doesn't seem to exist
 
+## Search all Support Content
```
Comment (lkurz) [inline 404.md:8]: Search looks good.  
Not sure why "This page doesnâ€™t seem to exist" renders this way in Codespace and VSCode.  But it looks fine when the html is viewed from SUPPORT 

image

File: _layouts/default.html [hunk 2]
```diff
@@ -52,6 +52,12 @@
         </ul>
         {% endif %}
         -->
+
```
Comment (lkurz) [inline _layouts/default.html:55]: Search needs better formatting... doesn't fit in the column.
Move to Body?
image

**Comment by loricooperhdf** [comment]
I tried something different, different functionality for search as provided by Google. Fit may still be an issue but the search content it won't load in preview in github codespaces for me, and I can't tell if it works, either. Can you check on this @lkurz?

**Comment by lkurz** [comment]
> I tried something different, different functionality for search as provided by Google. Fit may still be an issue but the search content it won't load in preview in github codespaces for me, and I can't tell if it works, either. Can you check on this @lkurz?

When you configure the PSE configuration to send to a results page it asks you for the url to return the results to.  
Currently it is  https://support.hdfgroup.org/results.html. the results will go to that page and not the page of the codespace.  This results page does not exist yet on support.hdfgroup.org so you when you search from codespace the PSE sends to SUPPORT and you get a 404.

During testing you can change the PSE results url to your codespace results page and it works.

**Comment by loricooperhdf** [comment]
> During testing you can change the PSE results url to your codespace results page and it works.

Are you saying this works? I couldn't figure out what URL to grab to plugin for the temp results page. Tried this, no change in results... (https://github.com/loricooperhdf/HDFGroup.github.io/blob/47da40550d0e5fa09b301ba05a66c7356bcf9cac/results.html)

**Review by lkurz — APPROVED** [review]
Looking good now.

**Comment by lkurz** [comment]
Created search folder and index.md file.
Layout search now links to there.

**Pull request merged**